src/main/java/com/github/manolo8/darkbot/gui/MapDrawer.java

```diff
@@ -224,4 +266,72 @@ public double toGameLocationY(int screenY) {
             return (screenY / (double) getHeight()) * mapBounds.getHeight();
         }
     }
+
+    private static class BackgroundImgDownloaderTask extends SwingWorker<Image, Void> {
+        private static final Pattern MINIMAP_ID_PATTERN = Pattern.compile("id=\"(\\d+)\".*minimap=\"(\\d+)\"");
+        private static Map<Integer, Integer> mapBackgroundIds;
+
+        private final MapDrawer mapDrawer;
+        private final int mapId;
+
+        public BackgroundImgDownloaderTask(MapDrawer mapDrawer, int mapId) {
+            this.mapDrawer = mapDrawer;
+            this.mapId = mapId;
+        }
+
+        @Override
+        protected void done() {
+            try {
+                mapDrawer.backgroundImage = get();
```

Comment:
I think we should cache these in a local folder, create a "maps" or "map-images" folder inside the bot folder, and search for the image there first, and if not available then download it